src/modules/ipfs/ipfs.service.ts

```diff
@@ -26,4 +54,45 @@ export class IPFSService {
       return false;
     }
   }
+
+  /**
+   * Get claim from cluster. If claim isn't found tries to get from gateway
+   *
+   * @param cid Content identifier.
+   * @returns Stringified credential
+   */
+  public async get(cid: string): Promise<string> {
+    let claim: string;
+    const getFromCluster = this.didStoreCluster.get(cid);
+    const getFromInfura = this.didStoreInfura.get(cid);
+    try {
+      claim = await Promise.any([getFromCluster, getFromInfura]);
+    } catch (e) {
+      // TODO: catch this in DidService
+      throw new HttpException(`Claim ${cid} not found`, HttpStatus.NOT_FOUND);
+    }
+
+    await this.pinsQueue.add(
+      PIN_CLAIM_JOB_NAME,
+      JSON.stringify({ cid, claim })
+    );
+    return claim;
+  }
+
+  /**
+   * Saves credential on cluster
+   *
+   * @param credential Credential being persisted
+   * @returns CID of the persisted credential
+   */
+  public async save(credential: string): Promise<string> {
+    try {
+      return this.didStoreCluster.save(credential);
+    } catch (_) {
+      this.logger.warn(
+        `Error saving ${credential} in cluster. Saving in Infura`
+      );
+      return this.didStoreInfura.save(credential);
+    }
+  }
```

Comment:
@JGiter I wonder if we should continue to save in Infura as well as the cluster. 

In the PR description you mentioned that:
> claims are not actually pinned. After claim is stored it can be found only during some period of time, which is no more than a day

I am concerned that if we are pinning to cluster but the cluster persistence isn't reliable, then we may save the data but it will not reliably be available. What do you think?

Reply:
Yes, it make sense